src/yggdrasil/multicast.go

```diff
@@ -16,19 +15,16 @@ type multicast struct {
 	reconfigure chan chan error
 	sock        *ipv6.PacketConn
 	groupAddr   string
-	myAddr      *net.TCPAddr
-	myAddrMutex sync.RWMutex
+	listeners   map[string]*tcpListener
 }
 
 func (m *multicast) init(core *Core) {
 	m.core = core
 	m.reconfigure = make(chan chan error, 1)
+	m.listeners = make(map[string]*tcpListener)
```

Comment:
It's kind of ugly that we have listeners here and also in the link's tcp struct. I'm not sure there's a better alternative, but I don't really like it...

Reply:
The main reason for this is that the map in `tcp.go` is a `map[listenaddr]listener` whereas the map in `multicast.go` is a `map[interfacename]listener`. The duplication isn't ideal but since `tcp.go` and `multicast.go` are responsible for populating their own maps, I don't *believe* this should be a problem.